pkg/kubelet/kubelet_node_status.go

```diff
@@ -602,9 +602,10 @@ func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error
 	return err
 }
 
-func (kl *Kubelet) isUpdateStatusPeriodExperid() bool {
+func (kl *Kubelet) isUpdateStatusPeriodExpired() bool {
+	// Ensure that everything is properly initialized if the node status has never been reported.
 	if kl.lastStatusReportTime.IsZero() {
-		return false
+		return true
 	}
```

Comment:
so this is the problem, if we remove it , we keep exactly the same logic, the thing is that now is more obvius.

Can you please also expand on the comment to explain the problem in more detail and reference the bug you opened?

```
If the node.status is never updated after a kubelet restart, then the periodic update loop is never executed
```

something like that

Reply:
Sure! Updated :slightly_smiling_face: